list2/ex50/functions/checkOut.js

```diff
@@ -0,0 +1,18 @@
+import promptSync from "prompt-sync";
+const prompt = promptSync();
+
+export function checkOut(hotels, reservations) {
+  console.log("\nInsert the reserve id to check out.");
+  const reservationId = parseInt(prompt(">"));
+  const foundReservation = reservations.find((reservation) => reservation.reservationId === reservationId);
+
+  if (foundReservation && foundReservation.checkOut()) {
+    const findingHotel = hotels.find((hotel) => hotel.id === foundReservation.reservationId);
+    if (findingHotel) {
+      findingHotel.vacateRoom();
+      console.log(`- Check out reservation id: [${reservationId}] successful`);
+    }
+  } else {
+    console.log(`FAILED! Reservation id: [${reservationId}] could not be check out.`);
+  }
+}
```

Comment:
You could try to avoid that nesting by applying the early return pattern we talked about in previous exercises.